opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -322,18 +322,29 @@ def get_current_span(self):
         """See `opentelemetry.trace.Tracer.get_current_span`."""
         return self._current_span_slot.get()
 
-    @contextmanager
     def start_span(
         self,
         name: str,
         parent: trace_api.ParentSpan = trace_api.Tracer.CURRENT_SPAN,
         kind: trace_api.SpanKind = trace_api.SpanKind.INTERNAL,
-    ) -> typing.Iterator[trace_api.Span]:
+    ) -> "Span":
         """See `opentelemetry.trace.Tracer.start_span`."""
 
         span = self.create_span(name, parent, kind)
         span.start()
-        with self.use_span(span, end_on_exit=True):
+        return span
+
+    @contextmanager
+    def start_as_current_span(
+        self,
+        name: str,
+        parent: trace_api.ParentSpan = trace_api.Tracer.CURRENT_SPAN,
+        kind: trace_api.SpanKind = trace_api.SpanKind.INTERNAL,
+    ) -> typing.Iterator[trace_api.Span]:
+        """See `opentelemetry.trace.Tracer.start_as_current_span`."""
+
+        span = self.start_span(name, parent, kind)
```

Comment:
Actually the function can be implemented more simply and efficiently without `@contextmanager` and `Iterator` by replaing the `with` statement: `return self.use_span(span, True)`.

Reply:
Hah! Right. I remember this is how I envisioned it first, but I guess I was decaffeinated when I wrote it ;(